# HG changeset patch
# User Rik <rik@octave.org>
# Date 1399067122 25200
#      Fri May 02 14:45:22 2014 -0700
# Node ID f8203daed9d45523384e281d47cf6f1fd40de245
# Parent  700714c099a22a26c2849ff2702e3d74b6287a6b
Abort Octave GUI if setsid() fails after fork().

* main.in.cc (main): Add in missing else keyword to restore warning and fail
behavior if setsid() does not work.

diff --git a/src/main.in.cc b/src/main.in.cc
--- a/src/main.in.cc
+++ b/src/main.in.cc
@@ -598,18 +598,18 @@ main (int argc, char **argv)
           // Child.
 
           if (setsid () < 0)
             {
               std::cerr << "octave: error calling setsid!" << std::endl;
 
               retval = 1;
             }
-
-          retval = octave_exec (file, new_argv);
+          else
+            retval = octave_exec (file, new_argv);
         }
       else
         {
           // Parent.  Forward signals to child while waiting for it to exit.
 
           int status;
 
           while (true)
